model_analyzer/record/record_aggregator.py

```diff
@@ -63,16 +78,23 @@ def filter_records(self, record_types=None, filters=None):
 
         Returns
         -------
-        Dict of List of Records
-            keys are record_types, values
-            are all records of that type
+        RecordAggregator
```

Comment:
If this function needs to have this type of functionality, we should think about making it static.

Reply:
The goal with this was to make the return types such that the return values can be chained together.

Reply:
Is it okay that a member function returns a whole new aggregator? I agree that we do need filter_records to return a `RecordAggregator` however,  could this be achieved just by passing in a RecordAggregator?
```
"""
@staticmethod
def filter_records(aggregator, record_types=None, filters=None):
        """
        Get records that satisfy the given list of criteria.
        Parameters
        ----------
        aggregator : The RecordAggregator
            containing records to be filtered
        record_types : list of types of Records
            the types of the records we are
            imposing the filter criteria on.
        filters : list of callables
            conditions that determine whether
            a given record should be returned.
            If no filters specified, all records
            of types specified by record_types will be
            returned.
            Note : This must be of the same length
                   as the list of record_types, or omitted.
        Returns
        -------
        RecordAggregator
            Returns a new RecordAggregator containing the filtered
            records
        """
        
        input_records = aggregator.get_records()
        filtered_records = RecordAggregator()
        if not record_types and not filters:
            for record_type, records in input_records.items():
                filtered_records.add_key(record_type, records)
            return filtered_records

        if record_types and not filters:
            try:
                for record_type in record_types:
                    filtered_records.add_key(record_type,
                                             input_records[record_type])
                return filtered_records
            except KeyError as k:
                raise TritonModelAnalyzerException(
                    f"Record type '{k.header()}' not found in this RecordAggregator"
                )
        if filters and not record_types:
            raise TritonModelAnalyzerException(
                "Must specify the record types corresponding to each filter criterion."
            )
        if len(record_types) != len(filters):
            raise TritonModelAnalyzerException(
                "Must specify the same number of record types as filter criteria."
            )

        # Remove records that do not satisfy criteria
        for h, f in zip(record_types, filters):
            for record in input_records[h]:
                if f(record):
                    filtered_records.insert(record)

        return filtered_records
```

Reply:
I agree that returning a whole new aggregator is not a good solution. I can add variants of these member functions that perform the filtering in place but I think making them static and passing the object as an argument may break the encapsulations that we have.